doc/userguide/configuration/suricata-yaml.rst

```diff
@@ -1584,6 +1584,96 @@ computers etc.)
 Packet Acquisition
 ------------------
 
+Data Plane Development Kit (DPDK)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+`Data Plane Development Kit <https://www.dpdk.org/>`_ consists of libraries to accelerate packet processing
+workloads running on a wide variety of CPU architectures.
+DPDK `Environment Abstraction Layer (EAL) <https://doc.dpdk.org/guides/prog_guide/env_abstraction_layer.html>`_
+provides a generic interface to low-level resources. It is a unique way how DPDK libraries access
+NICs. EAL creates an API for application to access NIC resources from the userspace level. In DPDK, packets
+are not retrieved via interrupt handling. Instead, the application
+`polls <https://doc.dpdk.org/guides/prog_guide/poll_mode_drv.html>`_ NIC for newly received packets.
+
+EAL allows the user space application to directly access memory where NIC stores the packets.
+As a result, neither DPDK nor the application copies the packets for the inspection. The application directly
+processes packets via passed packet descriptors.
+
+
+.. figure:: suricata-yaml/dpdk.png
+    :align: center
+    :alt: DPDK basic architecture
+    :figclass: align-center
+
+    `High-level overview of DPDK application`
+
+
+Suricata makes use of DPDK for packet acquisition in workers runmode.
+The whole DPDK configuration resides in the `dpdk:` node. This node encapsulates
+2 main subnodes and those are eal-params and interfaces.
+
+::
+
+    dpdk:
+        eal-params:
+            l: 0-1
+            a: 0000:3b:00.0
+        interfaces:
+            - interface: 0000:3b:00.0 # PCIe address of the NIC port
```

Comment:
@srini38 there are much easier ways to include virtual devices. But this really depends upon the `configure.ac` prepared by @lukashino.

I have already pointed out for the current edit in `configure.ac`  this will not work,

Reply:
@lukashino I recommend to check DPDK pkg-config (> 20.11) or makefile (< 20.11) how the libraries are linked. Becuase if you use pkg-config (> 20.11) or Makefile (<20.11) you current statement `work is focused to support physical machines and physical devices only so far` is incorrect.

Reply:
I have reviewed Makefiles of DPDK examples, I changed the LIBS / CFLAGS acquisition to:
CFLAGS="${CFLAGS} `pkg-config --cflags "libdpdk >= 20.11"`"
LIBS="${LIBS} `pkg-config --libs "libdpdk >= 20.11"`"